lightningd/gossip_control.c

```diff
@@ -554,6 +554,66 @@ static const struct json_command addgossip_command = {
 };
 AUTODATA(json_command, &addgossip_command);
 
+static struct command_result *json_listprivateinbound(struct command *cmd,
```

Comment:
Instead of a new API, perhaps we need to enhance listpeerchannels?  (optional) remote_private_update object inside that maybe?

In that case, we need to clean it up once channel is public (probably a good idea anyway!)  so gossipd needs to tell is when it (first) makes one of our private channels public (i.e. first sees a valid channel_announcement).  Note this can fire more than once (lost gossip store!) so don't freak out if we didn't have private info (could happent if they never sent before announcement anyway!).